chain/ethereum/src/data_source.rs

```diff
@@ -187,7 +187,7 @@ impl DataSource {
                 .iter()
                 .find(move |handler| {
                     handler.filter.is_some()
-                        && handler.filter.clone().unwrap() == BlockHandlerFilter::Call
+                        && handler.filter.as_ref().unwrap() == &BlockHandlerFilter::Call
```

Comment:
Lets refactor this whole condition to `handler.filter == Some(BlockHandlerFilter::Call)`

Reply:
Good point. Not sure why I didn't write it like that right away. Fixed now